### PR TITLE
Add table turbot_grant closes #12

### LINKED ISSUE #12: Add table turbot_grant
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

I would like to audit who has active grant permissions in my turbot workspace.

### PR DESCRIPTION
**Pull request by ParthaI**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select id, identity_status, ientity_family_name,level_trunk_title from turbot_active_grant
+-----------------+-----------------+---------------------+-------------------------------------------------------+
| id              | identity_status | ientity_family_name | level_trunk_title                                     |
+-----------------+-----------------+---------------------+-------------------------------------------------------+
| 161525342357056 | Active          | Smyth               | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 229155810239578 | Active          | Behera              | SuperUser                                             |
| 228825782849182 | Active          | Chakraborty         | SuperUser                                             |
| 216704470305117 | Active          | Das                 | User > Metadata > ReadOnly > Operator > Admin         |
| 232513631394343 | Active          | Behera              | User > Metadata > ReadOnly > Operator                 |
| 240323569425816 | Active          | Behera              | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 165127150940754 | Active          | Kumbha              | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 228825782907553 | Active          | Chakraborty         | User > Metadata > ReadOnly                            |
| 232514598645748 | Active          | Bhardwaj            | Role                                                  |
| 240323894253012 | Active          | Misra               | SuperUser                                             |
| 162687942600473 | Active          | Hadianto            | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 188609043625517 | Active          | Wallace             | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 189875991944587 | Active          | Bhardwaj            | SuperUser                                             |
| 162654220386929 | Active          | Bruno               | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 187951536899783 | Active          | Mohanty             | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 189793356024239 | Active          | Bhardwaj            | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 165127150928460 | Active          | Basu                | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 162199372503739 | Active          | Burgess             | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 240323832580553 | Active          | Misra               | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 162654220385904 | Active          | Bruno               | User > Metadata > ReadOnly > Operator > Admin         |
| 232513631215140 | Active          | Behera              | User > Metadata > ReadOnly > Operator                 |
| 161241086036962 | Active          | Admin               | User > Metadata > ReadOnly > Operator > Admin > Owner |
| 187955963611958 | Active          | Mohanty             | SuperUser                                             |
| 241556185269454 | Inactive        | Nayak               | User > Metadata                                       |
| 212996033743000 | Active          | Khuntia             | User > Metadata > ReadOnly > Operator > Admin         |
+-----------------+-----------------+---------------------+-------------------------------------------------------+
```
</details>

### CODE REVIEW

File: turbot/turbot_active_grant.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		Description: "All active grants of resources by Turbot.",
+		List: &plugin.ListConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "id", Require: plugin.Optional},
+			},
+			Hydrate: listActiveGrants,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the grant."},
```
Comment (rajlearner17) [inline turbot/turbot_active_grant.go:26]:
```suggestion
			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the grant."},
```
grant > grantee

**Comment by rajlearner17** [comment]
@ParthaI I would suggest having a `turbot_grant` table to capture grantee with status (Active/Inactive) and associated grants (Permissions)

File: turbot/table_turbot_grant.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+		List: &plugin.ListConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "id", Require: plugin.Optional},
+			},
+			Hydrate: listGrants,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the grantee."},
+			{Name: "grantee_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Status"), Description: "Status of the grantee."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:27]:
```suggestion
			{Name: "identity_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Status"), Description: "Status of the grantee."},
```

File: turbot/table_turbot_grant.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "grant_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Status"), Description: "Status of the grant."},
+			{Name: "display_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.DisplayName"), Description: "Display name of the grantee."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:29]:
```suggestion
			{Name: "identity_display_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.DisplayName"), Description: "Display name of the grantee."},
```

File: turbot/table_turbot_grant.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "email", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Email"), Description: "Email identity for the grantee."},
+			{Name: "family_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.FamilyName"), Description: "Family name of the grantee."},
+			{Name: "given_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.GivenName"), Description: "Given name of the grantee."},
+			{Name: "grant_identity_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant identity."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:33]:
```suggestion
			{Name: "identity_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant identity."},
```

File: turbot/table_turbot_grant.go [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "grant_type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant type."},
+			{Name: "grant_type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.URI"), Description: "URI of the grant type."},
+			{Name: "last_login_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Identity.LastLoginTimestamp"), Description: "Last login timestamp for the login."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:36]:
```suggestion
			{Name: "identity_last_login_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Identity.LastLoginTimestamp"), Description: "Last login timestamp for the login."},
```

File: turbot/table_turbot_grant.go [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "grant_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Status"), Description: "Status of the grant."},
+			{Name: "display_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.DisplayName"), Description: "Display name of the grantee."},
+			{Name: "email", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Email"), Description: "Email identity for the grantee."},
+			{Name: "family_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.FamilyName"), Description: "Family name of the grantee."},
+			{Name: "given_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.GivenName"), Description: "Given name of the grantee."},
+			{Name: "grant_identity_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant identity."},
+			{Name: "grant_type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant type."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:34]:
```suggestion
			{Name: "type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant type."},
```

File: turbot/table_turbot_grant.go [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "grant_type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.URI"), Description: "URI of the grant type."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:35]:
```suggestion
			{Name: "type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.URI"), Description: "URI of the grant type."},
```

File: turbot/table_turbot_grant.go [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "last_login_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Identity.LastLoginTimestamp"), Description: "Last login timestamp for the login."},
+			{Name: "level_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.Title"), Description: "The title of the level."},
+			{Name: "level_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.Trunk.Title"), Description: "Full title (including ancestor trunk) of the level."},
+			{Name: "level_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.URI"), Description: "The URI of the level."},
+			{Name: "profile_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.ProfileID"), Description: "Profile id of the grantee."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:40]:
```suggestion
			{Name: "identity_profile_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.ProfileID"), Description: "Profile id of the grantee."},
```

File: turbot/table_turbot_grant.go [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+			{Name: "resource_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Trunk.Title"), Description: "Full title (including ancestor trunk) of the resource."},
+			{Name: "resource_type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Type.URI"), Description: "URI of the resource type."},
+			{Name: "akas", Type: proto.ColumnType_JSON, Transform: transform.FromField("Identity.Akas"), Description: "AKA (also known as) identifiers for the grantee"},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:43]:
```suggestion
			{Name: "identity_akas", Type: proto.ColumnType_JSON, Transform: transform.FromField("Identity.Akas"), Description: "AKA (also known as) identifiers for the grantee"},
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see suggestions

General guidelines:
- Convert the field name into snake case
- Omit the word `turbot` in the column name
- Sometimes add the word `grant` or `active_grant` if there's ambiguity with another column name that's very close, e.g., `identity_status` vs. `grant_status`

File: turbot/table_turbot_grant.go [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+			Hydrate: listActiveAndInactiveGrants,
+		},
+		Columns: grantColumns(),
+	}
+}
+
+func grantColumns() []*plugin.Column {
+	return []*plugin.Column{
+		// Top columns
+		{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the grantee."},
```
Comment (rajlearner17) [inline turbot/table_turbot_grant.go:31]:
```suggestion
		{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the identity."},
```

Comment (rajlearner17) [reply turbot/table_turbot_grant.go:31]: change grantee > identity

File: turbot/table_turbot_grant.go [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+		{Name: "identity_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Status"), Description: "Status of the grantee."},
+		// {Name: "grant_status", Type: proto.ColumnType_STRING, Transform: transform.FromField("Status"), Description: "Status of the grant."},
+		{Name: "identity_display_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.DisplayName"), Description: "Display name of the grantee."},
+		{Name: "identity_email", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Email"), Description: "Email identity for the grantee."},
+		{Name: "ientity_family_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.FamilyName"), Description: "Family name of the grantee."},
+		{Name: "identity_given_name", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.GivenName"), Description: "Given name of the grantee."},
+		{Name: "identity_last_login_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Identity.LastLoginTimestamp"), Description: "Last login timestamp for the login."},
```
Comment (rajlearner17) [inline turbot/table_turbot_grant.go:38]:
```suggestion
		{Name: "identity_last_login_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Identity.LastLoginTimestamp"), Description: "Last login timestamp."},
```

File: turbot/table_turbot_grant.go [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+		{Name: "resource_type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Type.URI"), Description: "URI of the resource type."},
+		{Name: "identity_akas", Type: proto.ColumnType_JSON, Transform: transform.FromField("Identity.Akas"), Description: "AKA (also known as) identifiers for the grantee"},
+		// Other columns
+		{Name: "create_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.CreateTimestamp").NullIfEqual(""), Description: "The create time of grant."},
+		{Name: "filter", Type: proto.ColumnType_STRING, Transform: transform.FromQual("filter"), Description: "Filter used for this grant list."},
+		{Name: "timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.Timestamp").NullIfEqual(""), Description: "Timestamp when the grant was last modified (created, updated or deleted)."},
+		{Name: "update_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.UpdateTimestamp"), Description: "When the tag grant last updated in Turbot."},
```
Comment (rajlearner17) [inline turbot/table_turbot_grant.go:53]:
```suggestion
		{Name: "update_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.UpdateTimestamp"), Description: "When the grant was last updated in Turbot"},
```

File: turbot/table_turbot_grant.go [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+		{Name: "level_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.URI"), Description: "The URI of the level."},
+		{Name: "type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant type."},
+		{Name: "type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.URI"), Description: "URI of the grant type."},
+		{Name: "resource_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Trunk.Title"), Description: "Full title (including ancestor trunk) of the resource."},
+		{Name: "resource_type_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Type.URI"), Description: "URI of the resource type."},
+		{Name: "identity_akas", Type: proto.ColumnType_JSON, Transform: transform.FromField("Identity.Akas"), Description: "AKA (also known as) identifiers for the grantee"},
+		// Other columns
+		{Name: "create_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.CreateTimestamp").NullIfEqual(""), Description: "The create time of grant."},
```
Comment (rajlearner17) [inline turbot/table_turbot_grant.go:50]:
```suggestion
		{Name: "create_timestamp", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Turbot.CreateTimestamp").NullIfEqual(""), Description: "The create time of the grant."},
```

File: docs/tables/turbot_grant.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
+# Table: turbot_grant
+
+A grant is the assignment of a permission to a Turbot user or group on a resource or resource group. 
+
+## Examples
```
Comment (cbruno10) [inline docs/tables/turbot_grant.md:5]: We should include 3-4 more useful examples

File: turbot/table_turbot_grant.go [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+			{Name: "identity_profile_id", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.ProfileID"), Description: "Profile id of the identity."},
+			{Name: "identity_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Identity.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant identity."},
+			{Name: "level_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.Title"), Description: "The title of the level."},
+			{Name: "level_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.Trunk.Title"), Description: "Full title (including ancestor trunk) of the level."},
+			{Name: "level_uri", Type: proto.ColumnType_STRING, Transform: transform.FromField("Level.URI"), Description: "The URI of the level."},
+			{Name: "type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Type.Trunk.Title"), Description: "Full title (including ancestor trunk) of the grant type."},
```
Comment (cbruno10) [inline turbot/table_turbot_grant.go:39]: Why does this column and the one below not include the word `resource`, but the others below it do with similar property paths?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see comments, thanks!

A lot of comments are around field naming, we had established some guidelines in https://github.com/turbot/steampipe-plugin-turbot/pull/15#pullrequestreview-825513447, was there a reason you deviated from these here?

Also, I think there's a similar column naming issue I didn't include in this review, but did for active_grant here https://github.com/turbot/steampipe-plugin-turbot/pull/24#discussion_r782269061.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**